source/UF/ConnectedTypes.lagda

```diff
@@ -0,0 +1,223 @@
+                            ↔ is-surjection f
+ one-connected-map-iff-surj = (one-connected-map-is-surj , surj-is-one-connected-map)
+
+\end{code}
+
+We now devlop some closure conditions pertaining to connectedness. Connectedness
+is closed under equivalence as expected, but more importantly connectedness
+extends below: more precisely if a type is k connected then it is l connected
+for all l ≤ k. We provide many incarnations of this fact below which may prove
+useful. 
+
+\begin{code}
+
+ connectedness-closed-under-equiv : {X : 𝓤 ̇} {Y : 𝓥 ̇} {k : ℕ}
+                                  → X ≃ Y
+                                  → Y is k connected
+                                  → X is k connected
+ connectedness-closed-under-equiv e Y-con =
+   equiv-to-singleton (truncation-closed-under-equiv e) Y-con
+
+ contractible-types-are-connected : {X : 𝓤 ̇} {n : ℕ}
+                                  → is-contr X
+                                  → X is n connected
+ contractible-types-are-connected {𝓤} {X} {n} (c , C) = ((∣ c ∣[ n ]) , C')
+  where
+   C' : (s : ∥ X ∥[ n ]) → (∣ c ∣[ n ]) ＝ s
+   C' = ∥∥ₙ-ind (hlevels-are-closed-under-id ∥∥ₙ-hlevel (∣ c ∣[ n ]))
+                (λ x → ap ∣_∣[ n ] (C x))
+
+ connectedness-is-lower-closed : {X : 𝓤 ̇} {k : ℕ}
+                               → X is (succ k) connected
+                               → X is k connected
+ connectedness-is-lower-closed {𝓤} {X} {k} X-succ-con =
+   equiv-to-singleton successive-truncations-equiv 
+                      (contractible-types-are-connected X-succ-con)
+
+ connectedness-extends-to-zero : {X : 𝓤 ̇} {k : ℕ}
+                               → X is k connected
+                               → X is 0 connected
+ connectedness-extends-to-zero {𝓤} {X} {0} X-con = X-con
+ connectedness-extends-to-zero {𝓤} {X} {succ k} X-con =
+   connectedness-extends-to-zero (connectedness-is-lower-closed X-con)
+
+ connectedness-step-down : {X : 𝓤 ̇} {k l : ℕ}
+                         → X is (l +' k) connected
+                         → X is l connected
+ connectedness-step-down {𝓤} {X} {0} {l} X-con = X-con
+ connectedness-step-down {𝓤} {X} {succ k} {l} X-con =
+   connectedness-step-down (connectedness-is-lower-closed X-con)
+
+ connectedness-extends-below : {X : 𝓤 ̇} {k l : ℕ}
+                             → (l ≤ℕ k)
+                             → X is k connected
+                             → X is l connected
+ connectedness-extends-below {𝓤} {X} {k} {l} o X-con =
+   connectedness-step-down (transport (λ z → X is z connected) p X-con)
+  where
+   m : ℕ
+   m = pr₁ (subtraction l k o)
+   p = k        ＝⟨ pr₂ (subtraction l k o) ⁻¹ ⟩
+       m +' l   ＝⟨ addition-commutativity m l ⟩
+       l +' m   ∎
+
+\end{code}
+
+We can characterize connected types in terms of inhabitedness and connectedness
+of the identity type at the level below. We will assume univalence when neccesary.
+
+\begin{code}
+
+ conn-to-inhabited : {X : 𝓤 ̇} {n : ℕ}
```

Comment:
I would call this `inhabited-if-connected` and there is a more conceptual proof using `inhabited-if-one-connected`.

Reply:
Okay I'll check it out